dashboard/test/integration/account_purger_test.rb

```diff
@@ -0,0 +1,29 @@
+require 'test_helper'
+
+# Here, we want to ensure that User records can be purged via
+# the AccountPurger class.
+class AccountPurgerIntegrationTest < ActionDispatch::IntegrationTest
+  NULL_STREAM = File.open File::NULL, 'w'
+
+  setup do
+    # Never actually upload logs to S3
+    PurgedAccountLog.any_instance.stubs(:upload)
+  end
+
+  # We can purge a simple but typical student account, in general.
+  test 'can purge a typical student account' do
+    user = create :young_student_with_teacher
+
+    purger = AccountPurger.new(log: NULL_STREAM)
+    purger.purge_data_for_account(user)
```

Comment:
What exactly is being tested here?

Reply:
There are already unit tests to cover the purger's actual functionality... so I was not concerned with it much. This is an integration test 🤢 to, heh, effectively show that on-delete hooks elsewhere don't assume that a `user` exists to catch those kinds of regressions. (These prevent users from being deleted... which can be required with haste to settle a legal situation.) Though, the regression we have is also behind a flag, that might not be the case in the future... so... a general test case.

Integration tests are, to me, generally used to prove an integration regression is fixed. You add them to cover a bug case that a unit test didn't and cannot cover... and then remove them after a successful deploy. So this test really does not do much other than prove the experiment flag is the precedent. Without the flag stubbed the general integration test passes _before and after_ the fix. The following test added _fails before the fix and then passes afterward_.

It would, then, be reasonable to just delete these integration tests after the deploy. Technically, I can remove this integration test after the deploy and resolution of the purge queue. It just serves to prove the fix works in the production environment. However, the integration test is still useful for catching regressions. We should probably keep it as a simple precaution. Therefore, I can remove the stubbed test and leave the general test after the deploy.

So, what is a better integration test for the general case? It doesn't really need to assert anything that the unit tests don't already cover. I'm trying to anticipate future situations where hooks use a `user` that would act as `nil` because it is in a deleted state. Yet, I think I can settle some concerns to at least show that the user has a saved `purged_at` attribute so the test asserts something truthful rather than just a lack of some raised error.